### PR TITLE
Change publishing plugin

### PR DESCRIPTION
**Pull request by Thorinwasher**
# Description
I think we will be using a different publishing platform to maven central when changing domains, so we have tp also modify the publishing software. I also added the new maintainers of this project while I was at it.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [n/a] Unit tests added (if applicable).

### CODE REVIEW

**Comment by Thorinwasher** [comment]
Messes up publishing to maven local, looking into it

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -26,31 +26,12 @@ jobs:
...
-        uses: DamianReeves/write-file-action@v1.3
-        if: ${{ github.ref_type == 'tag' }}
-        with:
-          path: private.key
-          contents: ${{ secrets.PRIV_KEY }}
-          write-mode: overwrite
-
       - name: Publish
         if: ${{ github.ref_type == 'tag' }}
+        run: ./gradlew publishAndReleaseToMavenCentral --no-configuration-cache
```
Comment (thelooter) [inline .github/workflows/publish.yml]:
```suggestion
        run: ./gradlew publishAndReleaseToMavenCentral --no-configuration-cache -Pmockbukkit.version=$(echo ${{ github.ref_name }} | sed -e 's:v::')
```

Comment (thelooter) [reply .github/workflows/publish.yml]: We still need to set the version property

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -26,31 +26,12 @@ jobs:
...
-            -PossrhUsername=Iu0XBMzI \
-            -PossrhPassword=$OSSRH_PASSWORD \
-            -Psigning.secretKeyRingFile=private.gpg \
-            -Psigning.keyId=${{ secrets.PRIV_KEY_ID }} \
-            -Psigning.password=$SIGN_KEY
+          ORG_GRADLE_PROJECT_mavenCentralUsername: ${{ secrets.OSSRH_USERNAME }}
+          ORG_GRADLE_PROJECT_mavenCentralPassword: ${{ secrets.OSSRH_TOKEN }}
+          ORG_GRADLE_PROJECT_signingInMemoryKeyPassword: ${{ secrets.KEY_PASS }}
+          ORG_GRADLE_PROJECT_signingInMemoryKey: ${{ secrets.PRIV_KEY }}
+          REWRITE_VERSION: ${{ github.event.release.tag_name }}
```
Comment (thelooter) [inline .github/workflows/publish.yml]:
```suggestion
```

We don't need the Variable here, as we set the version property in the command above

File: build.gradle.kts [hunk 3]
```diff
@@ -122,61 +120,57 @@ java {
...
+				name.set("Eve Kolb")
+			}
+			developer{
+				id.set("thorinwasher")
+				name.set("Hjalmar Gunnarsson")
+				email.set("officialhjalmar.gunnarsson@outlook.com")
+			}
+			developer{
+				id.set("4everTheOne")
+				name.set("Afonso")
```
Comment (thelooter) [inline build.gradle.kts]: @4everTheOne do you want your full name in here too?

File: build.gradle.kts [hunk 4]
```diff
@@ -122,61 +120,57 @@ java {
...
+				id.set("thebusybiscuit")
+				name.set("TheBusyBiscuit")
+			}
+			developer {
+				id.set("insprill")
+				name.set("Pierce Thompson")
+			}
+			developer {
+				id.set("thelooter")
+				name.set("Eve Kolb")
```
Comment (thelooter) [inline build.gradle.kts:153]:
```suggestion
				id.set("thelooter")
				name.set("Eve Kolb")
				email.set("me@thelooter.de")
```

File: build.gradle.kts [hunk 5]
```diff
@@ -122,61 +120,57 @@ java {
...
+				email.set("sebastiaan.de.schaetzen@gmail.com")
+			}
+			developer {
+				id.set("thebusybiscuit")
+				name.set("TheBusyBiscuit")
+			}
+			developer {
+				id.set("insprill")
+				name.set("Pierce Thompson")
+			}
```
Comment (thelooter) [inline build.gradle.kts]: @Insprill do you want your email in here?

**Review by thelooter — CHANGES_REQUESTED** [review]

**Comment by thelooter** [comment]
We still need to update the Repository Secrets when we're ready to merge this

File: build.gradle.kts (hunk 5 above)
Comment (Insprill) [reply build.gradle.kts]: Sure

File: build.gradle.kts [hunk 6]
```diff
@@ -122,61 +120,58 @@ java {
...
+				email.set("me@thelooter.de")
+			}
+			developer{
+				id.set("thorinwasher")
+				name.set("Hjalmar Gunnarsson")
+				email.set("officialhjalmar.gunnarsson@outlook.com")
+			}
+			developer{
+				id.set("4everTheOne")
+				name.set("Afonso")
```
Comment (4everTheOne) [inline build.gradle.kts]:
```suggestion
				name.set("Afonso Oliveira")
```

File: build.gradle.kts (hunk 3 above)
Comment (4everTheOne) [reply build.gradle.kts:162]: Yes please

**Review by 4everTheOne — APPROVED** [review]
I'm a noob with this part of publishing to Maven Central :D
But within the knowledge I have, it looks good to me!

**Pull request merged**